tests/ballerina-integration-test/src/test/resources/auth/authservices/06_mock-oauth-service.bal

```diff
@@ -14,52 +14,391 @@
+                    if (param.contains("grant_type")) {
+                        grantType = param.split("=")[1];
+                    } else if (param.contains("scope")) {
+                        scopes = param.split("=")[1];
+                    } else if (param.contains("username")) {
+                        username = param.split("=")[1];
+                    } else if (param.contains("password")) {
+                        password = param.split("=")[1];
+                    }
+                }
+                res = prepareResponse(res, grantType, scopes, username, password, bearer);
+                _ = caller->respond(res);
+            } else {
+                // Invalid request. (Refer: https://tools.ietf.org/html/rfc6749#section-5.2)
+                res.statusCode = http:BAD_REQUEST_400;
+                json errMsg = { "error": "invalid_request" };
+                io:println(errMsg);
+                res.setPayload(errMsg);
+                _ = caller->respond(res);
+            }
+        }
     }
 
     @http:ResourceConfig {
-        methods: ["POST"]
+        methods: ["POST"],
+        path: "/token/refresh"
     }
-    resource function token(http:Caller caller, http:Request req) {
-        // Mock token refresh resource
-        var payload = req.getTextPayload();
-        if (payload is string) {
-            boolean clientIdInBody = payload.contains("client_id");
-            string authHeader;
-            var reqHeader = trap req.getHeader("Authorization");
-            if (reqHeader is error) {
-                authHeader = "";
+    // This refresh the access token but does not issue a new refresh token. The new access
+    // token will the MD5 hash of client_id + client_secret + refresh_token + scopes.
+    resource function refresh(http:Caller caller, http:Request req) {
+        http:Response res = new;
+        var authorizationHeader = trap req.getHeader("Authorization");
+        if (authorizationHeader is string) {
+            string clientIdSecret = CLIENT_ID + ":" + CLIENT_SECRET;
+            string expectedAuthorizationHeader = "Basic " + encoding:encodeBase64(clientIdSecret.toByteArray("UTF-8"));
+            if (authorizationHeader == expectedAuthorizationHeader) {
+                var payload = req.getTextPayload();
+                if (payload is string) {
+                    string[] params = payload.split("&");
+                    string grantType = "";
+                    string refreshToken = "";
+                    string scopes = "";
+                    foreach string param in params {
+                        if (param.contains("grant_type")) {
+                            grantType = param.split("=")[1];
+                        } else if (param.contains("refresh_token")) {
+                            refreshToken = param.split("=")[1];
+                            // If refresh token contains `=` symbol, we need to concat all the parts of the value since
```

Comment:
```suggestion
                            // If the refresh token contains the `=` symbol, then it is required to concatenate all the parts of the value since
```